File: Java/Fizzbuzz.java
/**
*Solution to kattis problem source: https://open.kattis.com/problems/fizzbuzz
*/

import java.util.Scanner;

public class Fizzbuzz {
  private int x, y;
  static int n;

  public void fizzbuzz(int number){
    String toPrint = "";
    if (number % x == 0) {
      toPrint += "Fizz";
    }
    if (number % y == 0){
      toPrint += "Buzz";
    }
    if (toPrint.length() == 0) {
      System.out.println(number);
    } else {
      System.out.println(toPrint);
    }
  }
  public void handleInput() {
    Scanner kattio = new Scanner(System.in);
    x = kattio.nextInt();
    y = kattio.nextInt();
    n = kattio.nextInt();
    kattio.close();
  }
  public static void main(String[] args){
    Fizzbuzz thisInstance = new Fizzbuzz();
    thisInstance.handleInput();
    for (int i=1; i <= n; i++){
      thisInstance.fizzbuzz(i);
    }
  }

}

File: Go/fizzbuzz.go
package main

import "fmt"

// Solution for Kattis: https://open.kattis.com/problems/fizzbuzz

// fizzBuzz loops through 1 -> c printing appropriate string for each iteration
func fizzBuzz(a, b, c int) {
	for i := 1; i <= c; i++ {
		str := ""
		if i%a == 0 {
			str += "Fizz"
		}
		if i%b == 0 {
			str += "Buzz"
		}
		if len(str) != 0 {
			fmt.Println(str)
		} else {
			fmt.Println(i)
		}
	}
}

func in() {
	var a, b, c int
	fmt.Scanf("%d", &a)
	fmt.Scanf("%d", &b)
	fmt.Scanf("%d", &c)
	fizzBuzz(a, b, c)
}

func main() {
	in()
}
